File: Data Structures & Algorithms/subarray-sum-equals-k/submission-3.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        // Track how many times each prefix sum has occured
        Map<Integer, Integer> map = new HashMap<>();
        int count = 0;
        int sum = 0;
        // Sum of 0 happens once (1 frequency)
        // To correctly count subarrays that start from index 0
        map.put(0, 1);

        for (int num : nums) {
            sum += num;
            count += map.getOrDefault(sum - k, 0);
            map.put(sum, map.getOrDefault(sum, 0) + 1);
        }
        return count;
    }
}

File: Data Structures & Algorithms/remove-element/submission-3.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        count = 0 # number of elements equal to val
        w = 0
        for num in nums:
            if num != val:
                nums[w] = num
                w += 1
            else:
                count += 1
        return len(nums) - count # number of elements not equal to val